### Task1:
You need to code a program that prints numbers from 1 to 20 except multiple of 2 & 3.

Hint: use "modulo" operator and "and/or" operators

In [ ]:
for a in range(1,21):
  if (a%2!=0 and a%3!=0):
    print(a)

1
5
7
11
13
17
19


### Task2:
Write a program that keep on accepting number from the user until user enters Zero. Display the sum and average of all the numbers.

Hint: Use while loop with flag.

In [ ]:
numbers= []
sum = 0
flag = True
while flag:
   number = int(input("enter numbers :"))
   if number==0:
       flag = False
   else:
    numbers.append(number)
    sum += number

if len(numbers)>0:
  length = len(numbers)
  average = sum / length
  print("The sum of given numbers is :",sum)
  print("The lenth of given numbers is :",length)
  print("The average of given numbers is :",average)
else:
  print("no number entered.")

enter numbers :4
enter numbers :5
enter numbers :6
enter numbers :0
The sum of given numbers is : 15
The lenth of given numbers is : 3
The average of given numbers is : 5.0


In [ ]:
type(number)

int

In [ ]:
type(numbers)

list

In [ ]:
type(len(numbers))

int

3### Task3:
Accept 10 numbers from the user and display their average.

Hint: Use loop for accepting numbers or use split to take input in one go

In [ ]:
nums = input("Enter 10 numbers for average :").split(',')
sum = 0
n = 0

for num in nums:
  sum+=int(num)
  n = len(nums)

print("sum of numbers :",sum)
print("total numbers :",n)
average = sum/n
print(f"The average of {nums} is :", average)

Enter 10 numbers for average :1,2,3,4,5,6,7,8,9,10
sum of numbers : 55
total numbers : 10
The average of ['1', '2', '3', '4', '5', '6', '7', '8', '9', '10'] is : 5.5


### Task4:
Write a program to display sum of odd numbers and even numbers that fall between 12 and 37(including both numbers)


In [ ]:
even_nums = 0
odd_nums = 0
for num in range(12,38):
  if num%2==0:
    even_nums += num

  else:
    odd_nums += num

print("sum of even numbers from 12 to 37 is :", even_nums)
print("sum of odd numbers from 12 to 37 is :",odd_nums)

sum of even numbers from 12 to 37 is : 312
sum of odd numbers from 12 to 37 is : 325


### Task5:
Write a program to print table of a number accepted from user.


In [ ]:
n1 = int(input("Enter a number for table :"))

for num in range(1,11):
  print(f"{n1} * {num} =",num*n1)

Enter a number for table :7
7 * 1 = 7
7 * 2 = 14
7 * 3 = 21
7 * 4 = 28
7 * 5 = 35
7 * 6 = 42
7 * 7 = 49
7 * 8 = 56
7 * 9 = 63
7 * 10 = 70


### Task6:
Write a program to display product of the digits of a number accepted from the user.

If user inputs:123 Outputs 6

If user inputs: 567 Outputs 210

In [ ]:
nums = input("Enter numbrs for product :").split(',')
product = 1
for num in nums:
  product *=int(num)
print(product)

Enter numbrs for product :5,6,7
210


### Task7:
Write a program to print first 10 even numbers in reverse order.

In [ ]:
for num in range(20,0,-2):
  print(num)

20
18
16
14
12
10
8
6
4
2


### Task8:
Write a program to find the sum of the digits of a number accepted from user

If user inputs:123 Outputs 6

If user inputs: 567 Outputs 18

In [ ]:
digits = input("Enter digits for sum :").split(',')
sum = 0
for digit in digits:
  sum +=int(digit)
print(sum)

Enter digits for sum :5,6,7
18


### Task9:
Write a program that accepts a sentence and calculate the number of letters and digits.

Suppose the following input is supplied to the program:

hello world! 123

Then, the output should be:

LETTERS 10

DIGITS 3


In [ ]:
sentence = input("Enter a sentence")
letter_count = 0
digit_count  = 0

for char in sentence:
  if char.isalpha():
    letter_count  += 1
  elif char.isdigit():
    digit_count  += 1
print("letters",letter_count )
print("digits",digit_count )


Enter a sentencehello world! 123
letters 10
digits 3


### Task10:
You need to trace and make a list of all such numbers from1000 to 3000 in which all the digits are even numbers.

In [ ]:
even_numbers =[]
for num in range(1000,3000,2):
  even_numbers.append(num)
print(even_numbers)


[1000, 1002, 1004, 1006, 1008, 1010, 1012, 1014, 1016, 1018, 1020, 1022, 1024, 1026, 1028, 1030, 1032, 1034, 1036, 1038, 1040, 1042, 1044, 1046, 1048, 1050, 1052, 1054, 1056, 1058, 1060, 1062, 1064, 1066, 1068, 1070, 1072, 1074, 1076, 1078, 1080, 1082, 1084, 1086, 1088, 1090, 1092, 1094, 1096, 1098, 1100, 1102, 1104, 1106, 1108, 1110, 1112, 1114, 1116, 1118, 1120, 1122, 1124, 1126, 1128, 1130, 1132, 1134, 1136, 1138, 1140, 1142, 1144, 1146, 1148, 1150, 1152, 1154, 1156, 1158, 1160, 1162, 1164, 1166, 1168, 1170, 1172, 1174, 1176, 1178, 1180, 1182, 1184, 1186, 1188, 1190, 1192, 1194, 1196, 1198, 1200, 1202, 1204, 1206, 1208, 1210, 1212, 1214, 1216, 1218, 1220, 1222, 1224, 1226, 1228, 1230, 1232, 1234, 1236, 1238, 1240, 1242, 1244, 1246, 1248, 1250, 1252, 1254, 1256, 1258, 1260, 1262, 1264, 1266, 1268, 1270, 1272, 1274, 1276, 1278, 1280, 1282, 1284, 1286, 1288, 1290, 1292, 1294, 1296, 1298, 1300, 1302, 1304, 1306, 1308, 1310, 1312, 1314, 1316, 1318, 1320, 1322, 1324, 1326, 1328, 1330, 133

In [ ]:
type(even_numbers)

list

### Task11:
You need to write a program that counts the number of vowels in a given sentence as input from console.

Suppose the following input is supplied to the program:

Hello world! Practice makes perfect

Then, the output should be:

a appeared 2 times

e appeared 5 times

i appeared  1 time

o appeared 2 times

u appeared 0 time


In [ ]:
sentence = input("Enter a senence :")
a = sentence.count("a")
e = sentence.count("e")
i = sentence.count("i")
o = sentence.count("o")
u = sentence.count("u")
print(f"a appeared {a} times")
print(f"e appeared {e} times")
print(f"i appeared {i} times")
print(f"o appeared {o} times")
print(f"u appeared {u} times")

Enter a senence :Hello world! Practice makes perfect
a appeared 2 times
e appeared 5 times
i appeared 1 times
o appeared 2 times
u appeared 0 times


### Task12:
You need to write a program that takes sequence of lines as input and prints the lines after making all characters in the sentence capitalized.

Suppose the following input is supplied to the program:

Hello world

Practice makes perfect

Then, the output should be:

HELLO WORLD
PRACTICE MAKES PERFECT

In [ ]:
a = input("Enter a sentence :")
b = input("Enter a sentence :")

x = a.upper()
y = b.upper()
print(x,y)



Enter a sentence :hello world !
Enter a sentence : practice make you perfect
HELLO WORLD !  PRACTICE MAKE YOU PERFECT


### Task13:
You need to write a code that takes a comma separated sequence of words as input and prints the words in a comma-separated sequence after sorting them alphabetically.

Suppose the following input is supplied to the program:

without,hello,bag,world

Then, the output should be:

bag,hello,without,world

In [ ]:
words = input("Enter words on your choice :").split(',')
words.sort()
print(words)

Enter words on your choice :ahsan,mohsin,ali,riaz,ahmed
['ahmed', 'ahsan', 'ali', 'mohsin', 'riaz']


### Task14:
You need code a program that calculates and prints the value according to the given formula:
P = Square root of [(2 * A * B)/C]

Following are the fixed values of A and B:

A is 50. B is 30.

The values of the literal C should be taken as console input to your program in a comma-separated sequence.

Example:
Let us assume the following comma separated input sequence is given to the program:
100,150,180

The output of the program should be:
18,22,24

In [ ]:
A = 50
B = 30
C = input("Enter a number :").split(',')
for num in C:
  C = int(num)
  p = (2 * A * B)/C
  print(int(p), sep=',')

Enter a number :100,150,180
30
20
16


### Task15:
You need to write a ocde in python that finds all such numbers which are divisible by 7 but are not a multiple of 5,
between 2000 and 3200 (both included).

Print the obtained numbers in a comma-separated sequence on a single line

Hint: modulo operator, and/or operator, for loop /while loop

In [ ]:
for num in range(2000,3200):
  if (num%7==0 and num%5!=0):
    print(num, end=',')

2002,2009,2016,2023,2037,2044,2051,2058,2072,2079,2086,2093,2107,2114,2121,2128,2142,2149,2156,2163,2177,2184,2191,2198,2212,2219,2226,2233,2247,2254,2261,2268,2282,2289,2296,2303,2317,2324,2331,2338,2352,2359,2366,2373,2387,2394,2401,2408,2422,2429,2436,2443,2457,2464,2471,2478,2492,2499,2506,2513,2527,2534,2541,2548,2562,2569,2576,2583,2597,2604,2611,2618,2632,2639,2646,2653,2667,2674,2681,2688,2702,2709,2716,2723,2737,2744,2751,2758,2772,2779,2786,2793,2807,2814,2821,2828,2842,2849,2856,2863,2877,2884,2891,2898,2912,2919,2926,2933,2947,2954,2961,2968,2982,2989,2996,3003,3017,3024,3031,3038,3052,3059,3066,3073,3087,3094,3101,3108,3122,3129,3136,3143,3157,3164,3171,3178,3192,3199,

### Task16:
With a given integral number n, write a program to generate a dictionary that contains (i, i*i) such that is an integral number between 1 and n (both included). and then the program should print the dictionary.

Suppose the following input is supplied to the program:

8

Then, the output should be:

{1: 1, 2: 4, 3: 9, 4: 16, 5: 25, 6: 36, 7: 49, 8: 64}

In [ ]:
#dic ={i:i*i}
n=int(input("Enter a number :"))
dic ={}
for num in range(1,n+1):
  dic[num] = num*num
print(dic)


Enter a number :8
{1: 1, 2: 4, 3: 9, 4: 16, 5: 25, 6: 36, 7: 49, 8: 64}


### Task17:
You need to write a code which accepts a sequence of comma separated 4 digit binary numbers as its input and then check whether they are divisible by 5 or not.

The numbers that are divisible by 5 are to be printed in a comma separated sequence.

Example:

0100,0011,1010,1001

Then the output should be:
1010

Reference:
1) https://www.datacamp.com/.../python-data-type-conversion

2) https://stackoverflow.com/questions/21765779/converting-binary-to-decimal-integer-output?fbclid=IwAR3DD_oiq5pn99IjyxSJgojvB5W6M4JqQtv4EUl_IF_DtmuRXlP-DswcKAA

In [ ]:
binary_sequence = input("Enter a sequence of comma-separated 4-digit binary numbers: ").split(',')

divisible_by_5 = []

# Check divisibility and store numbers divisible by 5
for binary_num in binary_sequence:
    decimal_num = int(binary_num, 2)
    print(decimal_num)
    if decimal_num % 5 == 0:
        divisible_by_5.append(binary_num)

# Print the result
print(','.join(divisible_by_5))

Enter a sequence of comma-separated 4-digit binary numbers: 1010,0101,1001,0110,100
10
5
9
6
4
1010,0101


In [ ]:
for num in range(5):
  numbers = int(input("enter a number :"))
  #n = numbers.map(",")
  #for num in numbers:
  if (num%5==0):
    print(numbers)

In [ ]:
total = 0
for a in range(10):
  numbers = int(input("Enter numbers for sum :"))
  total +=numbers
print("sum of numbers is :",total)

Enter numbers for sum :1
Enter numbers for sum :2
Enter numbers for sum :3
Enter numbers for sum :4
Enter numbers for sum :5
Enter numbers for sum :6
Enter numbers for sum :7
Enter numbers for sum :8
Enter numbers for sum :9
Enter numbers for sum :10
sum of numbers is : 55


In [ ]:
sum = 0
[f"sum of even is {sum+=num} is even" if num%2==0 else f"sum of odd is {sum+=num}" for num in range(12,38) ]

In [ ]:
evens_sum = 0
odds_sum = 0

evens = [num for num in range(12,38) if num%2==0 f"{num.sum(num)}"]
odds = [num for num in range(12,38) if num%2!=0]
print(evens)
print(odds)

evens_sum += num

In [ ]:
[f"{num} is even" if num%2==0 else f"{num} is odd" for num in range(12,38) ]

['12 is even',
 '13 is odd',
 '14 is even',
 '15 is odd',
 '16 is even',
 '17 is odd',
 '18 is even',
 '19 is odd',
 '20 is even',
 '21 is odd',
 '22 is even',
 '23 is odd',
 '24 is even',
 '25 is odd',
 '26 is even',
 '27 is odd',
 '28 is even',
 '29 is odd',
 '30 is even',
 '31 is odd',
 '32 is even',
 '33 is odd',
 '34 is even',
 '35 is odd',
 '36 is even',
 '37 is odd']

In [ ]:
for table_num in range(2,5):
  for num in range (1,11):
    print(f"{table_num} * {num} = {num * table_num}")

2 * 1 = 2
2 * 2 = 4
2 * 3 = 6
2 * 4 = 8
2 * 5 = 10
2 * 6 = 12
2 * 7 = 14
2 * 8 = 16
2 * 9 = 18
2 * 10 = 20
3 * 1 = 3
3 * 2 = 6
3 * 3 = 9
3 * 4 = 12
3 * 5 = 15
3 * 6 = 18
3 * 7 = 21
3 * 8 = 24
3 * 9 = 27
3 * 10 = 30
4 * 1 = 4
4 * 2 = 8
4 * 3 = 12
4 * 4 = 16
4 * 5 = 20
4 * 6 = 24
4 * 7 = 28
4 * 8 = 32
4 * 9 = 36
4 * 10 = 40
